Automattic/wp-calypso
# Theme Discovery: Fix more style variations button not working

In the Discovery screen, we have some themes in collections that have more than the maximum displayed style variations. When this happens a + icon is shown and when clicking on it we should navigate to the theme details page.

This PR amends that error so that our users can see more style variations when click on the corresponding icon on a theme card that's inside a Collection Carousel. 

Related to:
- https://github.com/Automattic/dotcom-forge/issues/4380

Builds on:
- https://github.com/Automattic/wp-calypso/pull/83456

## Proposed Changes

* Routed to the theme details page (just as done in the rest of the Theme Showcase).

## Testing Instructions



* Load the Calypso Live Link
* Go to the LoTS and add `?flags=themes/discovery-lots` 
* On the main Discover view find a theme card that has more style variations than what can be displayed.
* Click on the `+X` button
* You should navigate to the theme details page. 

Before:
https://github.com/Automattic/wp-calypso/assets/1989914/37c27126-9709-4152-8955-b440ea2b193c

After:
https://github.com/Automattic/wp-calypso/assets/1989914/2647ff8a-07c0-4eac-842e-f9126a9f0fef


## Pre-merge Checklist
